wolfcrypt/src/hash.c

```diff
@@ -1706,3 +1706,32 @@ int wc_HashGetFlags(wc_HashAlg* hash, enum wc_HashType type, word32* flags)
 
 #endif /* !NO_HASH_WRAPPER */
 
+#ifdef WOLFSSL_HASH_KEEP
+int _wc_Sha_Grow(byte** msg, word32* used, word32* len, const byte* in,
+                        int inSz, void* heap)
+{
+    if (*len < *used + inSz) {
+        if (*msg == NULL) {
+            *msg = (byte*)XMALLOC(*used + inSz, heap, DYNAMIC_TYPE_TMP_BUFFER);
+        }
+        else {
+            byte* pt = (byte*)XMALLOC(*used + inSz, heap,
```

Comment:
XREALLOC?

Reply:
Yeah, for some reason I was avoiding XREALLOC at the time of writing the code. I see no reason though to not switch it.